File: app/src/main/java/org/mifos/mobile/ui/views/BeneficiaryApplicationView.java
package org.mifos.mobile.ui.views;

import org.mifos.mobile.models.templates.beneficiary.BeneficiaryTemplate;
import org.mifos.mobile.ui.views.base.MVPView;

/**
 * Created by dilpreet on 16/6/17.
 */

public interface BeneficiaryApplicationView extends MVPView {

    void showUserInterface();

    void showBeneficiaryTemplate(BeneficiaryTemplate beneficiaryTemplate);

    void showBeneficiaryCreatedSuccessfully();

    void showBeneficiaryUpdatedSuccessfully();

    void showError(String msg);

    void setVisibility(int state);
}

File: app/src/main/java/org/mifos/mobile/ui/views/BeneficiaryApplicationView.kt
package org.mifos.mobile.ui.views

import org.mifos.mobile.models.templates.beneficiary.BeneficiaryTemplate
import org.mifos.mobile.ui.views.base.MVPView

/**
 * Created by dilpreet on 16/6/17.
 */
interface BeneficiaryApplicationView : MVPView {
    fun showUserInterface()
    fun showBeneficiaryTemplate(beneficiaryTemplate: BeneficiaryTemplate?)
    fun showBeneficiaryCreatedSuccessfully()
    fun showBeneficiaryUpdatedSuccessfully()
    fun showError(msg: String?)
    fun setVisibility(state: Int)
}